astropy/modeling/core.py

```diff
@@ -808,7 +808,8 @@ def outputs(self, val):
         self._outputs = val
 
     @property
-    def n_inputs(self):
+    def n_inputs(self):  # noqa: F811 # see https://github.com/astropy/astropy/pull/16633
```

Comment:
All the temporary Ruff configuration belongs in `.ruff.toml`. The configuration in `pyproject.toml` and all uses of `noqa` are meant to be permanent.

Reply:
I really cannot see both not make #16633 blocking and not have a noqa here. Any suggestion ?

Reply:
Add the rule to the `modeling` section in `.ruff.toml`. Whether we should update the code to enforce the rule or ignore the rule with noqas should be the outcome of #16633.

Reply:
done ! I rebased the branch to avoid back-and-forth in the history